From b4ec445e9fa763f5d510b56f78c55dd4ce01192b Mon Sep 17 00:00:00 2001
From: Moria Zuberi <moria.zuberi@tandemg.com>
Date: Thu, 12 May 2022 17:54:57 +0300
Subject: [PATCH] [SW-0] preboot: recovery: remove bin compilation

Remove recovery.bin compilation and intallation.
Originally this was added as preparetion for erom, but eventually
erom was added in diffrent folder and files, so recovery.bin is
redundant and unused.

Change-Id: I376e8808037c9efa9211202815a1e686e42ad58e
Signed-off-by: Moria Zuberi <moria.zuberi@tandemg.com>
---

diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index c644385..903a4cb 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -17,7 +17,6 @@
 PPBOOT_BINFILE=${PPBOOT_BINFILE:-ppboot.bin}
 PREBOOT_BINFILE=${PREBOOT_BINFILE:-preboot.bin}
 PCIE_PHY_BINFILE=${PCIE_PHY_BINFILE:-pcie_phy_fw.bin}
-RECOVERY_BINFILE=${RECOVERY_BINFILE:-recovery.bin}
 PIF_ARC_ELFFILE=${PIF_ARC_FILENAME:-pcie_pif.elf}
 
 ZEPHYR_PID_DIR=${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-custom/build-pid
@@ -297,9 +296,6 @@
 
 	#generate public key components
 	generate_public_keys
-
-	# Create recovery image
-	prep_image_recovery ${RECOVERY_BINFILE}
 
 	# Create pif arc binaries
 	prep_image_pcie_pif ${PIF_ARC_ELFFILE}
diff --git a/package/preboot/preboot.mk b/package/preboot/preboot.mk
old mode 100644
new mode 100755
index 2ad5448..f854a29
--- a/package/preboot/preboot.mk
+++ b/package/preboot/preboot.mk
@@ -18,11 +18,6 @@
 		$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D)/app/ext_rom all; \
 	fi;
 
-
-	if [ "$(BR2_PACKAGE_PREBOOT_ARC_RECOVERY)" == "y" ]; then \
-		$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D)/app/recovery all; \
-	fi;
-
 	if [ "$(BR2_PACKAGE_PCI_PIF_ARC)" == "y" ]; then \
 		$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D)/app/pcie_pif all; \
 	fi;
@@ -44,10 +39,6 @@
 		$(INSTALL) -D -m 0644 $(@D)/bin/ppboot.bin $(BINARIES_DIR)/ppboot.bin; \
 		$(INSTALL) -D -m 0644 $(@D)/bin/preboot.bin $(BINARIES_DIR)/preboot.bin; \
 		$(INSTALL) -D -m 0644 $(@D)/bin/pcie_phy_fw.bin $(BINARIES_DIR)/pcie_phy_fw.bin; \
-	fi;
-
-	if [ "$(BR2_PACKAGE_PREBOOT_ARC_RECOVERY)" == "y" ]; then \
-		$(INSTALL) -D -m 0644 $(@D)/app/recovery/recovery.bin $(BINARIES_DIR)/recovery.bin; \
 	fi;
 
 	if [ "$(BR2_PACKAGE_PREBOOT_ARC_EXT_ROM)" == "y" ]; then \
